knowledge_graphs/ai_script_analyzer.py

```
@@ -0,0 +1,550 @@
+            if isinstance(node, (ast.If, ast.While, ast.For, ast.Try, ast.With)):
+                complexity += 1
+            # Function/class definitions add complexity
+            elif isinstance(node, (ast.FunctionDef, ast.AsyncFunctionDef, ast.ClassDef)):
+                complexity += 2
+            # Nested structures add more complexity
+            elif isinstance(node, (ast.ListComp, ast.DictComp, ast.SetComp, ast.GeneratorExp)):
+                complexity += 1
+        
+        result["complexity_score"] = complexity
+    
+    def _detect_suspicious_patterns(self, tree: ast.AST, content: str, result: dict[str, Any]):
```

Ruff (0.12.2)

Missing return type annotation for private function `_detect_suspicious_patterns`

Add return type annotation: `None`

(ANN202)